### DISCUSSION TITLE
How to input particle size distribution measurements measurements in PK-sim?

### DISCUSSION THREAD

**Discussion by tobiasK2001** [Open-Systems-Pharmacology/Forum#819] [How-Tos] [answered]
Dear all,
I got these kind of particle measurements (laser difraction measurement with mass size calculation:  Mass percent (Volume distribution, density 1) from different compound milling processes:
![image](https://user-images.githubusercontent.com/25585248/98910727-d8d51d00-24c3-11eb-935f-8f069d2aa669.png)
However in PK-sim I can only input mean and SD (assuming normal distribution) or a geomean particle radius with a coefficient of variation (assuming log normal particle size distribution). 
Any Idea how can I used the measured mean, 10 and 90 % measurements in PK-Sim?

Thanks for your Input!

Best, Tobias

**Comment by Yuri05** [comment]
As far as I understand you have 3 predefined particle sizes. From your distribution of particles and the total dose it should be possible to calculate the dose for each particle size (however I don't quite understand how to interprete your inputs, so I cannot say how)

Then you could proceed like below:
1) Create 3 **monodisperse** particles formulations, one for each radius
2) Create user defined administration protocol with 3 single administrations (calculate 
<img width=700 src="https://user-images.githubusercontent.com/25061876/98917456-b1368280-24cc-11eb-86ac-c6454a52b6a8.png">
3) Map the 3 formulations in the simulation
<img width=700 src="https://user-images.githubusercontent.com/25061876/98917740-08d4ee00-24cd-11eb-914d-f0c89723bd29.png">

**Comment by AndreDlm** [comment, marked as answer]
Hi Tobias,

a few weeks ago I published a mechanistic IVIVC workflow using the particle dissolution model in OSP; you can find it [here](https://github.com/Open-Systems-Pharmacology/IVIVE-with-particle-dissolution-module-in-OSP).
In this repository you'll also find dummy `.json` files for the formulation and administration building block in PK-Sim; in the formulation json, you can specify the particle size for each bin and the in the administration json the drug amount in each of these bins is defined. The `R` script helps you to generate `json` files tailored to your specific data. 
I guess that will help you

Best,
André

**Reply by NingJia012** [reply]
Hi AndreDlm, thank you very much for your excellent work. I have a question regarding the mechanistic IVIVC framework. In the provided example, the DissolutionData.xlsx file is included. If I were to change the compound or formulation, would this file also need to be regenerated through new experiments? Additionally, if I lack in vitro dissolution data but only have particle size and cumulative fraction data, can your case study still be applied effectively?

**Reply by AndreDlm** [reply]
Dear @NingJia012, if you have different compound or formulation you need to populate the file `DissolutionData.xlsx` with your experimental dissolution data. If you do not have experimental data, but only particle size and fraction information, you can still apply the workflow but the simulations will be purely predictive and highly uncertain. It is therefore recommended to test the simulations with experimental data. 
Please note that we have recently published an updated and more mechanistic workflow which can be found [here](https://github.com/Open-Systems-Pharmacology/Oral-PBBM-Workflow)

**Comment by tobiasK2001** [comment]
Dear both,

thank you for the quick reply. @Juri I don't think it will work that way as I have particle distribution measurements, which I interpret as 
mean X 10 = 10 % of the number of all particles are smaller or equal to a size of  0.56 µm
mean X 50 = 50 %/median of the number of all particles are smaller or equal to a size of  2.01 µm
mean X 90 = 90 %/median of the number of all particles are smaller or equal to a size of  5.66 µm 
So I don't think I can translate that into 3 bins of particles. 
I was wondering if that can be translated to a mean and SD of the distribution (assuming normal distribution). But it seems that the shape is not symmetrically enough to allow that and it could be an oversimplification.
@André, looking at your input data (particle_distribution_raw_data.csv, cumulative distribution for 21 particle sizes) you used in your example it seems to be also not directly comparable to the measured output I have. My feeling now is that I have to get back to the CMC-guys who did the measurements and ask them if they can create some usable output from their measurements. It does not seem to be easy to get an standard way how to use particle size measurements. However, your workflow is a great step forward.

Thanks you both very much!

Best, Tobias

**Comment by Yuri05** [comment]
@tobiasK2001 Well, you need to make somehow an assumption about your particles radius distribution. For example:
* 10% of the particles have the radius `0.56 / 2` µm
* 40 % (=50-10) of particles have the radius `(0.56 + 2.01) / 2 `µm
* 40 % (=90-50) of the particles have the radius `(2.01 + 5.66) / 2` µm
* 10% of the particles have the radius `5.66` µm

Based on this you can define 4 monodisperse formulations, calculate the dose for each particle radius and proceed as suggested above. 
Or you can refine the binning above by linear interpolation (thus introducing N bins for each Interval [0 .. 0.56], [0.56 .. 2.01], ... and setting their radius via linear interpolation between min and max)

Of course if you have a better knowledge about the particles radius distribution, you can incorporate it into your binning.

**Comment by AndreDlm** [comment]
@tobiasK2001 You need to make an assumption about the distribution type (normal, lognormal etc) but using the script `psd calculation.R` in the [IVIVE repository](https://github.com/Open-Systems-Pharmacology/IVIVE-with-particle-dissolution-module-in-OSP) you can fit different distribution functions to your 3 data values and select the function that describes the data best. 
In the next step you use the fitted function to discretize into *n* bins (max. 10) and calculate for each bin the median radius (based on your fitted function) and save it as a `xlsx` file. 
Using the script `json.R` you finally generate PK-Sim compatible formulation and administration building blocks that contains for each bin the respective radius and drug amount, respectively.

**Comment by tobiasK2001** [comment]
Dear both,
Thanks a lot for the explanation! I will check it out!

Best, Tobias

**Comment by tobiasK2001** [comment]
Dear both,
now I run the workflow and it works like a charm :-). Very cool stuff indeed. Thanks a lot for this.
B.t.w. to load the jason files for BB in developer mode is  a super nice thing. When does it come in the "normal mode"?

Cheers, Tobias

**Comment by Yuri05** [comment]
> B.t.w. to load the jason files for BB in developer mode is a super nice thing. When does it come in the "normal mode"?

not before somebody created a feature request for that 😉

**Comment by tobiasK2001** [comment]
Done! May the gods of PK-Sim hear my pleading ;-)

**Comment by amitvtaneja** [comment]
Hi all

both approaches seem workable. How can either of these  be adapted for a modified release preparation intended for release in the colon? In other words i need to incorporate a lag time to the dissolution input.

Thanks in advance

Amit

**Comment by StephanSchaller** [comment]
Dear Amit (@amitvtaneja),

this is currently not available for the PSD formulation out-of-the-box in PK-Sim. Also fitting the particle sizes as described by @AndreDlm above to such a strongly delayed dissolution time will likely fail. You likely already know, but this can currently only be defined for the other formulations such as lint80 or Weibull. With some tools you can also easily characterize / fit the Weibull distribution based on a biorelevant dissolution test.
There might be options to delay the release of a PSD through adaptations of the PSD release model in MoBi.

Best,
Stephan

**Comment by Andr** [comment]
Why am I getting this email? I'm a bit confused

**Comment by StephanSchaller** [comment]
@AndreDlm , you received this (my last) comment also as email (automatically sent by GitHub) as you have replied in this (OSP GitHub) thread earlier and were mentioned in my comment.

**Comment by Andr** [comment]
Im not Andre my name is Andrew

**Comment by Yuri05** [comment]
@Andr You were wrongly referenced in a [comment above](https://github.com/Open-Systems-Pharmacology/Forum/issues/639#issuecomment-726023118), please ignore.

**Comment by amitvtaneja** [comment]
> Why am I getting this email? I'm a bit confused
> […](#)
> On Tue, Dec 8, 2020, 3:44 PM esqlabs GmbH ***@***.***> wrote: Dear Amit ***@***.*** <https://github.com/amitvtaneja>), this is currently not available for the PSD formulation out-of-the-box in PK-Sim. Also fitting the particle sizes as described by @AndreDlm <https://github.com/AndreDlm> above to such a strongly delayed dissolution time will likely fail. You likely already know, but this can currently only be defined for the other formulations such as lint80 or Weibull. With some tools you can also easily characterize / fit the Weibull distribution based on a biorelevant dissolution test. There might be options to delay the release of a PSD through adaptations of the PSD release model in MoBi. Best, Stephan — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub <[#639 (comment)](https://github.com/Open-Systems-Pharmacology/Forum/issues/639#issuecomment-741081826)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AOQ4WWFWLI34NQ5EFKD4YLDST2M4DANCNFSM4TS6AUXQ> .



> @AndreDlm , you received this (my last) comment also as email (automatically sent by GitHub) as you have replied in this (OSP GitHub) thread earlier and were mentioned in my comment.



> Dear Amit (@amitvtaneja),
> 
> this is currently not available for the PSD formulation out-of-the-box in PK-Sim. Also fitting the particle sizes as described by @AndreDlm above to such a strongly delayed dissolution time will likely fail. You likely already know, but this can currently only be defined for the other formulations such as lint80 or Weibull. With some tools you can also easily characterize / fit the Weibull distribution based on a biorelevant dissolution test.
> There might be options to delay the release of a PSD through adaptations of the PSD release model in MoBi.
> 
> Best,
> Stephan

Dear Stephan

I agree using a Weibull/Lint80 may be the most forthcoming option in PKsim. In my case, there seems to be a work-around however. Under admininstration protocol (user defined), the dose can be delivered directly to the target organ (lumen caecum). Perhaps this could work? PSD would bepreferred as it is mechanistic,

best regards

Amit
